torch/nn/modules/activation.py

```diff
@@ -741,19 +744,37 @@ def _reset_parameters(self):
     @weak_script_method
     def forward(self, query, key, value, key_padding_mask=None, incremental_state=None,
                 need_weights=True, static_kv=False, attn_mask=None):
-        """
-        Inputs of forward function
-            query: [target length, batch size, embed dim]
-            key: [sequence length, batch size, embed dim]
-            value: [sequence length, batch size, embed dim]
-            key_padding_mask: if True, mask padding based on batch size
-            incremental_state: if provided, previous time steps are cashed
-            need_weights: output attn_output_weights
-            static_kv: key and value are static
-
-        Outputs of forward function
-            attn_output: [target length, batch size, embed dim]
-            attn_output_weights: [batch size, target length, sequence length]
+        r"""
+    Args:
+        query, key, value: map a query and a set of key-value pairs to an output. 
+            See "Attention Is All You Need" for more details. 
+        key_padding_mask: if provided, specified padding elements in the key will 
+            be ignored by the attention.
+        incremental_state: if provided, previous time steps are cached.
+        need_weights: output attn_output_weights.
+        static_kv: if true, key and value are static. The key and value in previous 
+            states will be used.
+        attn_mask: mask that prevents attention to certain positions.
+
+    Shape:
+        - Inputs:
+
+        - query: :math:`(L, N, E)` where L is target length, N is batch size, E is 
+          embedding dimension.
+        - key: :math:`(S, N, E)`, ByteTensor, where S is sequence length, N is batch size, E is 
+          embedding dimension.
+        - value: :math:`(S, N, E)` where S is sequence length, N is batch size, E is 
+          embedding dimension.
+        - key_padding_mask: :math:`(N, S)` where N is batch size, S is sequence length.
+        - incremental_state: a dictionary used for storing states.
+        - attn_mask: :math:`(L, L)` where L is target length.
+
+        - Outputs:
+
+        - attn_output: :math:`(L, N, E)` where L is target length, N is batch size, 
+          E is embedding dimension.
+        - attn_output_weights: :math:`(N, L, S)` where N is batch size,
+          L is target length, S is sequence length.
```

Comment:
```suggestion
          L is the query sequence length, and S is the key sequence length.
```

Reply:
I would suggest to keep the "target" sequence and "source" sequence since they are not treated as the same in attention layer. In the attention layer, the query, as a target, is mapped with the key and value sequence.